src/core/chart.js

```diff
@@ -0,0 +1,252 @@
+  var monthlyData = [];
+  var timestamps = Object.keys(data).sort();
+  
+  if(timestamps.length == 17) {
+    throw new Error("Error: concatenated 17-point chronology.");
+  }
+  
+  if(timestamps.length != expectedTimestamps[timescale]) {
+    throw new Error("Error: inconsistent time resolution in data");
+  }
+  
+  for(var i = 0; i < 12; i++) {
+    var mapped = Math.ceil((timestamps.length / 12.0) * (i + 1)) - 1;
+    monthlyData.push(data[timestamps[mapped]]);
+  }
+  
+  //Seasonal timeseries need one month of winter removed from the beginning of the
+  //year and added at the end, since winter wraps around the calendar new year.
+  if(timescale == "seasonal") {
+    monthlyData = monthlyData.slice(1, 12);
+    monthlyData.push(data[timestamps[0]]);
+  }
+  
+  return monthlyData;
+};
+
+/*
+ * Helper function for parseTimeseriesForC3. Given a set of timeserieses 
+ * to be graphed and metadata about each timeseries, returns a function 
+ * that generates the shortest name necessary to distinguish a particular
+ * timeseries from all others being shown on the same chart.
+ * 
+ * For example, when graphing monthly, seasonal, and yearly means for 
+ * otherwise identical data run, only "monthly", "seasonal," and "yearly"
+ * need to appear in the graph legend. But if graphing multiple variables,
+ * the graph legend will need to display variable names as well.
+ * 
+ * Timeseries names include any descriptive  metadata that vary between 
+ * timeseries and leave out any metadata that doesn't. They end with "mean".
+ */
+//TODO: special case climatological period to display as (XXXX-XXXX)
+//TODO: possibly cue descriptors to appear in a specific order?
+// "Tasmin Monthly Mean" sounds better than "Monthly Tasmin Mean".
+var shortestUniqueTimeseriesNamingFunction = function (metadata, data) {
+  
+  var capitalizeWords = function(s) {
+    return s.replace(/\b\w/g, c => c.toUpperCase());
+  };
+  
+  //only one timeseries being graphed, simple label.
+  if(data.length == 1) {
+    return function(m) { return capitalizeWords(`${m.timescale} mean`)};
+  }
+  
+  var variation = [];
+  var exemplarMetadata = _.find(metadata, function(m) {return m.unique_id === data[0].id;});
+  
+  for(var i = 0; i < data.length; i++) {
+    var comparandMetadata = _.find(metadata, function(m) {return m.unique_id == data[i].id});
+    
+    for(var att in comparandMetadata) {
+      if(exemplarMetadata[att] !== comparandMetadata[att] && variation.indexOf(att) == -1) {
+        variation.push(att);
+      }
+    }
+  }
+  if(variation.length == 0) {
+    throw new Error("Error: cannot graph identical timeseries");
+  }
+  
+  //Remove unique_id from the list of possible variations. All
+  //datasets have unique unique_id's; it's not useful on a graph
+  variation.splice(variation.indexOf("unique_id"), 1);
+  
+  //Remove variable_name; it's redundant with variable_id (and takes
+  //up a lot of room.)
+  variation.splice(variation.indexOf("variable_name"), 1);
```

Comment:
There needs to be a check that `variable_name` is actually in `variation` before attempting to remove it. Unlike `unique_id`, `variable_name` is not always present.